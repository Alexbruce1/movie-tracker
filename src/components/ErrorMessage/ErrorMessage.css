h1 {
  margin: 0;
  padding: 0;
}
html, body, .app {
  height: 100vh;
  margin: 0;
  padding: 0;
  position: relative;
  width: 100%;
}
.ErrorMessage {
  background-image: url('../../assets/movies.jpg');
  height: 100vh;
  padding-top: 30vh;
  position: relative;
  width: 100%;
}

.ErrorMessage::after {
  content: '';
  background-color: #000;
  left: 0;
  height: 100%;
  opacity: 0.86;
  position: absolute;
  top: 0;
  width: 100%;
}

.ErrorMessage_inner {
  align-items: center;
  background-color: #f2f2f2;
  box-shadow: inset 0 0 6px rgba(0, 0, 0, .4);
  border-radius: 10px;
  color: #222;
  display: flex;
  flex-direction: column;
  height: 25rem;
  justify-content: center;
  padding: 5px;
  position: relative;
  margin: auto;
  width: 20%;
  min-width: 32rem;
  z-index: 1;
}

.ErrorMessage button {
  border-radius: 2px;
  background-color: #ff3030;
  border: none;
  box-shadow: 0 1px 1px rgba(0, 0, 0, .3);
  color: #fcfcfc;
  cursor: pointer;
  display: block;
  font-size: 18px;
  font-weight: 600;
  margin: 1rem auto;
  padding: 8px 1rem;
  outline: none;
  transition-duration: .2s;
  width: 24rem;
}

.home-link-error {
  color: #fcfcfc;
  text-decoration: none;
}