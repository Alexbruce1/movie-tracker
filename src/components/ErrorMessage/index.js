import React, { Component } from 'react';
import './ErrorMessage.css'
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

class ErrorMessage extends ComponentÂ {
  render() {
    const { message, closeError, user } = this.props

    return (
      <div className='ErrorMessage'>
        <div className='ErrorMessage_inner'>
          <h1 className='Login-header'> Movie Tracker </h1>
          <p>{message}</p>
          <button className='home-button-error'
            onClick={closeError}>
            <NavLink to='/' className='home-link-error'>Home
            </NavLink>
          </button>
          {!user.loggedIn &&
          <button className='login-button-error'
            onClick={closeError}>
            <NavLink to='/login' className='login-link-error'>Login/Create Account
            </NavLink>
          </button>
        }
        </div>
      </div>
    );
  }
}

ErrorMessage.propTypes = {
  message: PropTypes.string,
  closeError: PropTypes.func,
  user: PropTypes.object
}

export default ErrorMessage;
