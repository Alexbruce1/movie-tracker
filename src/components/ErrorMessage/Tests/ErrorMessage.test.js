import React from 'react';
import ErrorMessage from '../';
import { shallow } from 'enzyme';

describe('ErrorMessage', () => {
	let wrapper;
	beforeEach(() => {
    const mockUser = { id: 1, loggedIn: true }
		wrapper = shallow(<ErrorMessage closeError={jest.fn()} user={mockUser}/>)
	});

	it('Should render like snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
