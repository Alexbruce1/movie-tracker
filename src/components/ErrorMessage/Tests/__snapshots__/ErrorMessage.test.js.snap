// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage
    closeError={[MockFunction]}
    user={
      Object {
        "id": 1,
        "loggedIn": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ErrorMessage_inner"
      >
        <h1
          className="Login-header"
        >
           Movie Tracker 
        </h1>
        <p />
        <button
          className="home-button-error"
          onClick={[MockFunction]}
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="home-link-error"
            to="/"
          >
            Home
          </NavLink>
        </button>
      </div>,
      "className": "ErrorMessage",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="Login-header"
          >
             Movie Tracker 
          </h1>,
          <p />,
          <button
            className="home-button-error"
            onClick={[MockFunction]}
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="home-link-error"
              to="/"
            >
              Home
            </NavLink>
          </button>,
          false,
        ],
        "className": "ErrorMessage_inner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Movie Tracker ",
            "className": "Login-header",
          },
          "ref": null,
          "rendered": " Movie Tracker ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="home-link-error"
              to="/"
            >
              Home
            </NavLink>,
            "className": "home-button-error",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Home",
              "className": "home-link-error",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          "type": "button",
        },
        false,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ErrorMessage_inner"
        >
          <h1
            className="Login-header"
          >
             Movie Tracker 
          </h1>
          <p />
          <button
            className="home-button-error"
            onClick={[MockFunction]}
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="home-link-error"
              to="/"
            >
              Home
            </NavLink>
          </button>
        </div>,
        "className": "ErrorMessage",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="Login-header"
            >
               Movie Tracker 
            </h1>,
            <p />,
            <button
              className="home-button-error"
              onClick={[MockFunction]}
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="home-link-error"
                to="/"
              >
                Home
              </NavLink>
            </button>,
            false,
          ],
          "className": "ErrorMessage_inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Movie Tracker ",
              "className": "Login-header",
            },
            "ref": null,
            "rendered": " Movie Tracker ",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="home-link-error"
                to="/"
              >
                Home
              </NavLink>,
              "className": "home-button-error",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Home",
                "className": "home-link-error",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": "button",
          },
          false,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
